src/commands/copy/copy-worker.ts

```diff
@@ -79,145 +32,117 @@ export const worker = new WorkerRpc<CopyContract>({
       skipped: 0,
       skippedBytes: 0,
     };
-    const end = Math.min(args.start + args.size, args.manifest.length);
+    const statsUpdaters = {
```

Comment:
if you type this as `Reacord<FileOperation, (sourceSize, outputSize) => void>` you could drop all the types, 

also do these need to be objects it adds a fair bit of overhead

Reply:
also if this took a `stats` as a object, it makes it really easy to test

`(stats: Stats, sourceSize: number, outputSize:number) => void`